src/string.cr

```diff
@@ -2200,6 +2200,52 @@ class String
     match_result.try &.begin(0)
   end
 
+  # Searches sep or pattern (regexp) in the string from the end of the string,
+  # and returns the part before it, the match, and the part after it.
+  # If it is not found, returns two empty strings and str.
+  #
+  # ```
+  # "hello".rpartition("l")         #=> ["hel", "l", "o"]
+  # "hello".rpartition("x")         #=> ["", "", "hello"]
+  # "hello".rpartition(/.l/)        #=> ["he", "ll", "o"]
+  # ```
+  def rpartition(search : (Char | String)) : Tuple(String, String, String)
+    pos = self.rindex(search) || 0
+    search_size = search.is_a?(Char) ? 1 : search.size
+
+    if pos == 0 \
+      || pos == self.size-1 \
```

Comment:
there are some inconsistencies in the formatting. Use the crystal formatter tool otherwise the CI will complain.